gobblin-api/src/test/java/org/apache/gobblin/util/io/GsonInterfaceAdapterTest.java

```diff
@@ -43,4 +44,21 @@ public void test() {
 
   }
 
+  @Test
+  public void testObjectToIntegerDeserialize() {
+    Gson gson = new GsonBuilder().create();
+    Integer test = 5;
+    String ser = gson.toJson(test);
+    Object deser = gson.fromJson(ser, Object.class);
+    Assert.assertNotEquals(test, deser);
```

Comment:
we can also assert on the type of object returned(double) by default gson deserializer